### PR TITLE
fix : Crashing of reportValidity() for IE

### PR DESCRIPTION
**Pull request by jeffin143**
Handling : https://github.com/opencollective/opencollective/issues/2505

Thanks @Betree for the suggestion

### CODE REVIEW

**Comment by jeffin143** [comment]
> You included non-related commits in this PR, you should be able to get rid of them by rebasing to master.
> 
> image

Acknowledged, Sorry didn't notice it

Thanks for pointing out :)

File: components/CreateVirtualCardsForm.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ class CreateVirtualCardsForm extends Component {
   onSubmit(e) {
     e.preventDefault();
     const { values, submitting, deliverType } = this.state;
-    if (!submitting && this.form.current.reportValidity()) {
+    if (!submitting && this.form.current.reportValidity && this.form.current.reportValidity()) {
```
Comment (Betree) [inline components/CreateVirtualCardsForm.js]: We should not prevent submit if the method is not supported b the browser

```suggestion
    if (!submitting && (!this.form.current.reportValidity || this.form.current.reportValidity())) {
```

**Comment by Betree** [comment]
Thanks for rebasing :+1: 

Can we create a helper in a new file `lib/form-utils.js` do reduce the boilerplate code and prevent errors in the future?

```es6
/** This function will return true if reportValidity is not supported by the browser, or if it succeed */
export const reportValidityHTML5 = (domNodeOrEvent) => {
  return !domNodeOrEvent || !domNodeOrEvent.reportValidity || domNodeOrEvent.reportValidity();
}
```

File: components/contribution-flow/index.js [hunk 2]
```diff
@@ -333,7 +333,8 @@ class CreateOrderPage extends React.Component {
 
   /** Validate step profile, create new org if necessary */
   validateStepProfile = async () => {
-    if (!this.state.stepProfile || !this.activeFormRef.current || !this.activeFormRef.current.reportValidity()) {
+    // We don't want to crash if `reportValidity` is not supported by the browser
+    if (!this.state.stepProfile || !this.activeFormRef.current || !reportValidityHTML5(this.activeFormRef.current)) {
```
Comment (Betree) [inline components/contribution-flow/index.js]: Thanks to the new helper, we can simplify this line

```suggestion
    if (!this.state.stepProfile || !reportValidityHTML5(this.activeFormRef.current)) {
```

File: components/contribution-flow/index.js [hunk 3]
```diff
@@ -333,7 +333,8 @@ class CreateOrderPage extends React.Component {
 
   /** Validate step profile, create new org if necessary */
   validateStepProfile = async () => {
-    if (!this.state.stepProfile || !this.activeFormRef.current || !this.activeFormRef.current.reportValidity()) {
+    // We don't want to crash if `reportValidity` is not supported by the browser
```
Comment (Betree) [inline components/contribution-flow/index.js]: Let's remove the comment as it doesn't seems required anymore now that the check is abstracted by the helper 

```suggestion
```

File: components/contribution-flow/index.js [hunk 4]
```diff
@@ -622,7 +623,7 @@ class CreateOrderPage extends React.Component {
         name: 'details',
         isCompleted: Boolean(stepDetails && stepDetails.totalAmount >= minAmount),
         validate: () => {
-          return stepDetails && this.activeFormRef.current && this.activeFormRef.current.reportValidity();
+          return stepDetails && this.activeFormRef.current && reportValidityHTML5(this.activeFormRef.current);
```
Comment (Betree) [inline components/contribution-flow/index.js]: Thanks to the new helper, we can simplify this line

```suggestion
          return stepDetails && reportValidityHTML5(this.activeFormRef.current);
```

Comment (jeffin143) [reply components/contribution-flow/index.js]: Thanks :)

File: components/contribution-flow/index.js (hunk 3 above)
Comment (jeffin143) [reply components/contribution-flow/index.js]: yeah correct :)

**Pull request merged**